plugins/chat/lib/chat_message_rate_limiter.rb

```diff
@@ -37,7 +37,7 @@ def clear!
 
   def silence_user
     silenced_for_minutes = SiteSetting.chat_auto_silence_duration
-    return unless silenced_for_minutes > 0
+    return if silenced_for_minutes <= 0
```

Comment:
This already has min 0 on the setting, so this can be `return if silenced_for_minutes.zero?`